dluftspring/tap-hellobaton
# Script to Generate new Schemas and update `tap.py` based on a Swagger Specification

## Description & motivation

It is a pain to have to update the list of available schemas when new ones are launched. This PR implements a process to pull the json schemas from hellobatons swagger specification and then generate valid json schemas and update the sub streams in `tap.py` using python's abstract syntax tree. The only thing left to do is author integration tests and then run this process on a cron schedule so that the available streams never deviate from what is outlined in the swagger specification.

### This is a...

- [x] :tada: New feature
- [ ] :beetle: Bug fix
- [ ] :gear: Refactor
- [ ] :recycle: Housekeeping
  
## Screenshots

## Open questions

## How to test

Delete or alter a schema file in `schemas/` and delete the corresponding substream in `tap.py` then run 

```bash
poetry run swagger-sync
```

After that the schema you deleted should be re added and it's corresponding python code should be generated as well
